### ISSUE TITLE
[BUG] Setting fragment metadata causes errors

### ISSUE THREAD

**Issue by daveisfera** [awslabs/amazon-kinesis-video-streams-producer-c#396]
**Media pipeline:**
 - What is the media source (GStreamer, custom, etc). What is the upstream encoder/pipeline? Samples

**Logging**
[kvs_fragment_metadata_logs.txt](https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/files/12325635/kvs_fragment_metadata_logs.txt)

**Describe the bug**
When setting Fragment Metadata, errors are reported about invalid MKV data and then that the limit has been reached

**SDK version number**
1.4.1

**Open source building**
N/A

**To Reproduce**
1. Set fragment metadata
2. Send video frame
3. Observe errors

**Expected behavior**
Sending Fragment Metadata would not impact the video and would work correctly

**Desktop (please complete the following information):**
 - OS: Debian Bookworm
 - Browser: N/A
 - Version: N/A
 
**Additional context**
I [added this with the `develop` branch](https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/pull/394) but then I noticed that we were having an error when using the [current stable release (1.4.1)](https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/releases/tag/v1.4.1) and was able to reproduce the error with the sample

Basically, this appears to be fixed in `develop` already (or at least handled in a way that doesn't produce any error output), but I wanted to document that this was happening because it my be related to the issues we've been seeing in production

**Comment by daveisfera** [comment]
@jdelapla Following up on the [last question from #368 about the status code changing](https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/issues/368#issuecomment-1693914136), this appears to not be working and isn't covered under the expected constraints of the KVS Producer SDK, so is this a verified issue that needs to be fixed?

**Comment by jdelapla** [comment]
With the release of [1.5.0](https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/releases/tag/v1.5.0) on 8/25 develop changes are now present in the release. Changes in how the fragment metadata was placed to accommodate another KVS feature are likely responsible for the improvement in performance for  `putKinesisVideoFragmentMetadata` .
